src/System.Net.Primitives/src/System/Net/IPEndPoint.cs

```diff
@@ -58,7 +58,7 @@ public IPEndPoint(long address, int port)
             {
                 throw new ArgumentOutOfRangeException(nameof(port));
             }
-            _port = port;
+            _port = Convert.ToUInt16(port);
```

Comment:
This should just be `(ushort)port`.  It was already validated to be in range by ValidatePortNumber, and this Convert call just bloats things unnecessarily.  Same for the rest of the Convert calls below: they should just be casts.

Reply:
Done